src/app/dataset-view/additional-components/metadata-component/metadata.component.ts

```diff
@@ -117,6 +85,25 @@ export class MetadataComponent extends BaseComponent implements OnInit {
             });
     }
 
+    public getRouteLink(tab: MetadataTabs): string {
+        return `/${this.datasetMetadataTabData.datasetBasics.owner.accountName}/${this.datasetMetadataTabData.datasetBasics.name}/${DatasetViewTypeEnum.Metadata}/${tab}`;
+    }
+
+    public getVisibilityMenuItem(tab: MetadataTabs): boolean {
+        switch (tab) {
+            case MetadataTabs.PollingSource:
+                return Boolean(this.currentPollingSource);
+            case MetadataTabs.PushSources:
+                return Boolean(this.currentPushSources?.length);
+            case MetadataTabs.Transformation:
+                return Boolean(this.currentTransform);
+            case MetadataTabs.Watermark:
+                return this.isRoot;
```

Comment:
Is this fine for a case with push sources?

Reply:
Yes, fine for a case with push sources